moveit_ros/planning/planning_scene_monitor/src/planning_scene_monitor.cpp

```diff
@@ -531,18 +531,25 @@ void PlanningSceneMonitor::newPlanningSceneCallback(const moveit_msgs::PlanningS
 
 void PlanningSceneMonitor::clearOctomap()
 {
-  if (scene_->getWorldNonConst()->removeObject(scene_->OCTOMAP_NS))
-    triggerSceneUpdateEvent(UPDATE_SCENE);
-  if (octomap_monitor_)
+  bool removed = false;
   {
-    octomap_monitor_->getOcTreePtr()->lockWrite();
-    octomap_monitor_->getOcTreePtr()->clear();
-    octomap_monitor_->getOcTreePtr()->unlockWrite();
-  }
-  else
-  {
-    ROS_WARN_NAMED(LOGNAME, "Unable to clear octomap since no octomap monitor has been initialized");
+    boost::unique_lock<boost::shared_mutex> ulock(scene_update_mutex_);
```

Comment:
It wasn't hard to figure out how this works, but I had to think for a second because I don't use mutexes a lot. Do you want to add a comment to the effect of `// Lock scene using mutex before removing the octomap. Lock expires after the code block` for the less familiar? I'm on the fence.

Reply:
I should probably add a comment about the extra brackets that are there so that the scoped lock gets lifted before notifying consumers (`triggerSceneUpdateEvent`) in order to avoid a deadlock ...

Reply:
I didn't know that the lock needs to be lifted before that call. How about adding `// Lift the scoped lock before calling triggerSceneUpdateEvent to avoid deadlock` at the end of the block?

I'm fine with not adding the other comment because I suppose it's not "[nonobvious to a reader who understands C++ well]](https://google.github.io/styleguide/cppguide.html#Implementation_Comment_Donts)".